pkg/kube/controllers/webhook_configuration.go

```diff
@@ -166,27 +172,47 @@ func (f *WebhookConfig) generateValidationWebhookServerConfig(ctx context.Contex
 		},
 	}
 
-	for _, webhook := range webhooks {
-		ctxlog.Debugf(ctx, "Calculating validation webhook '%s'", webhook.Name)
-
-		url := url.URL{
-			Scheme: "https",
-			Host:   net.JoinHostPort(f.config.WebhookServerHost, strconv.Itoa(int(f.config.WebhookServerPort))),
-			Path:   webhook.Path,
+	if f.config.Provider == "gke" {
```

Comment:
could we only create `admissionregistrationv1beta1.WebhookClientConfig` for different cases and then assign it to webhook.ClientConfig?

Reply:
I wanted to do that, but the `Path` key in `WebhookClientConfig` depends on the for loop. So couldn't do that.

Reply:
Why, for performance reasons? I don't think creating a struct is especially expensive  in Go. I should benchmark that sometime...

Oh, because of the duplication with the else branch?
I'd try to  introduce a `func newWebhook(webhook, clientConfig) admissionregistrationv1beta1.Webhook` to dry this out

Reply:
Done. Thanks :)